src/utils.ts

```diff
@@ -447,3 +448,10 @@ export function createDestroyer(server: http.Server): () => Promise<void> {
     return destroyPromise;
   };
 }
+
+/**
+ * Indicates whether the end-user is running the CLI from a cloud-based environment.
+ */
+export function isCloudEnvironment() {
+  return !!env.CODESPACES;
```

Comment:
nit: for consistency with use elsewhere in the codebase, let's just do `process.env.CODESPACES`

Reply:
Done! 👍